readthedocs_build/config/config.py

```diff
@@ -205,6 +210,23 @@ def validate_base(self):
             base = validate_directory(base, base_path)
         self['base'] = base
 
+    def validate_build(self):
+        # Defaults
+        build = {'image': '2.0'}
+        if 'build' in self.raw_config:
+            _build = self.raw_config['build']
+            if 'image' in _build:
+                with self.catch_validation_error('build'):
+                    build['image'] = validate_choice(
+                        str(_build['image']),
+                        self.DOCKER_SUPPORTED_VERSIONS,
+                    )
+        self['build'] = build
+
+        # Set the valid python versions for this container
+        img = build['image']
+        self.PYTHON_SUPPORTED_VERSIONS = DOCKER_BUILD_IMAGES[img]['python']['supported_versions']
```

Comment:
`PYTHON_SUPPORTED_VERSIONS` is named as a constant and shouldn't be treated as a variable. I think the usage of `get_valid_python_versions` was more correct. Also, validation steps probably shouldn't have side effects like altering the state here, so that's probably also another good reason to make this a method that merges sources together.

Reply:
The ordering of the validation is going to continue to matter (because we need to read the docker image, then determine python version), so it will have to set state somewhere, but agreed on making it a bit nicer.

Reply:
I believe we are/were doing ordering of validation one level up anyways:
https://github.com/rtfd/readthedocs-build/blob/master/readthedocs_build/config/config.py#L133-L157

So, dependencies would naturally be filled there, as the validation steps are ordered there